docs/madr/decisions/011-mesh-traffic-route.md

```diff
@@ -0,0 +1,687 @@
+is the value from the `consumer` route:
+
+```yaml
+spec:
+ targetRef:
+  kind: MeshService
+  name: frontend
+ to:
+  - targetRef:
+     kind: MeshService
+     name: backend
+    rules:
+     - matches:
+        path:
+         prefix: /v2
+       filters:
+        - requestHeaderModifier:
+           add:
+            - name: env
+              value: dev
+       backendRefs:
+        - weight: 100
+          kind: MeshServiceSubset
+          name: backend
+          tags:
+           version: v2
+```
+
+This is likely surprising, especially because `consumer` didn't specify anything
+at all for `/v1`.
+
+This MADR proposes to distribute `default` down into individual `rules` and merge
+`rules` based on _structural equality_ of the `matches` value.
+So we would instead have:
+
+```yaml
+metadata:
+ name: owner
+spec:
+ targetRef:
+  kind: Mesh
+ to:
+  - targetRef:
+     kind: MeshService
+     name: backend
+    default:
+     rules:
+      - matches:
+         path:
+          prefix: /v1
+        backendRefs:
+         - weight: 100
+           kind: MeshServiceSubset
+           name: backend
+           tags:
+            version: v1
+      - matches:
+         path:
+          prefix: /v2
+        backendRefs:
+         - weight: 100
+           kind: MeshServiceSubset
+           name: backend
+           tags:
+            version: v2
+---
+metadata:
+ name: consumer
+spec:
+ targetRef:
+  kind: MeshService
+  name: frontend
+ to:
+  - targetRef:
+     kind: MeshService
+     name: backend
+    default:
+     rules:
+      - matches:
+         path:
+          prefix: /v2
+        filters:
+         - requestHeaderModifier:
+            add:
+             - name: env
+               value: dev
+        backendRefs:
+         - weight: 100
+           kind: MeshService
+           name: backend
+           tags:
+            version: v2
+```
+
+which gives us the rules:
+
+```yaml
+spec:
+ targetRef:
+  kind: MeshService
+  name: frontend
+ to:
+  - targetRef:
+     kind: MeshService
+     name: backend
+    rules:
+      - matches:
+         path:
+          prefix: /v1
+        backendRefs:
+         - weight: 100
+           kind: MeshServiceSubset
+           name: backend
+           tags:
+            version: v1
+     - matches:
+        path:
+         prefix: /v2
+       filters:
+        - requestHeaderModifier:
+           add:
+            - name: env
+              value: dev
+       backendRefs:
+        - weight: 100
+          kind: MeshService
+          name: backend
+          tags:
+           version: v2
+```
+
+Here, the `/v1` rule from `owner` is unchanged and the `/v2` rule from
+`consumer` overrides that of `owner`.
+
+#### `MeshGateway`
+
+This new resource is intended to replace the `MeshGatewayRoute` resource for
+configuring routes for a `MeshGateway`.
+
+There are two different ways we can express a `MeshGateway` attachment either as
```

Comment:
This is the decision to make. I'm not quite sure about either one, though I tend to prefer the latter. The latter is somewhat closer to how we're doing `targetRef` now but it may be confusing as to why sometimes we use `to` and sometimes `from`.

Reply:
I think we should stick to the idea "spec.targetRef selects proxy that will be configured". One of the flaws of the sources/destination approach we had is an inability to say where the conf is going to land without looking into the docs. I guess this means option "MeshGateway in spec.targetRef" for me

Reply:
> One of the flaws of the sources/destination approach we had is an inability to say where the conf is going to land without looking into the docs

But... assuming the mesh is working correctly, is this that important (for the route case)? Maybe better to continue this in slack